text/0000-doc-consts.md

```diff
@@ -0,0 +1,134 @@
+- Feature Name: `doc_consts`
+- Start Date: 2025-01-20
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+- Pre-RFC: [Pre-RFC: `#[doc(consts)]` attribute](https://internals.rust-lang.org/t/pre-rfc-doc-consts-attribute/21987)
+
+# Summary
+[summary]: #summary
+
+Introduce a `#[doc(consts = ...)]` attribute controlling how constant expressions are rendered by rustdoc.
+
+# Motivation
+[motivation]: #motivation
+
+Different crates and items have conflicting requirements for their constants.
+For some, [the exact value of a constant is platform dependant](https://internals.rust-lang.org/t/pre-rfc-doc-consts-attribute/21987/9).
+For others, [constant folding obsurces the meaning of values](https://github.com/rust-lang/rust/issues/128347).
+Hovever, [showing a constant as written may leak implementation details], 
+and in some cases, there is no possible value 
+
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The `#[doc(consts)]` attribute can be placed on any item to control how contained constant expressions are displayed in rustdoc-generated documentation.
+
+* `#[doc(consts = "fold")]` will show them in their fully-evaluated state.
```

Comment:
how about `"eval"` for fully evaluated item and "`as-is`" for item as written?

Reply:
"constant folding" is a well established term used for this exact process, and "eval" is much more general (future display modes would likely also involve evaluating the expression).

on the other hand, i don't feel strongly about `"expr"`, but my second choice would probably be `"verbatim"`.

Reply:
Doesn't Rust ever refer to CTFE as constant folding? The docs use evaluation https://doc.rust-lang.org/reference/const_eval.html, and to me "const folding" is an identify+transform optimization pass (as opposed to compile-time evaluation that must unconditionally be run).